### PR TITLE
Bug/watcher performance

### PR DESCRIPTION
**Pull request by tilt**
First steps to accelerate updates of work packages that have many watchers.

https://www.openproject.org/issues/2448

### CODE REVIEW

**Comment by tilt** [comment]
There are some failing tests to be fixed before merge due to delayed mail delivery - I've assigned myself to do this.

**Comment by tilt** [comment]
Fixed unit tests

**Comment by tilt** [comment]
Ticket changed to https://www.openproject.org/issues/2448

File: doc/CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ See doc/COPYRIGHT.rdoc for more details.
 
 # Changelog
 
+* `#2448` Accelerate work package updates
```
Comment (hschink) [inline doc/CHANGELOG.md:37]: Please respect the numerical order of those tickets in the changelog: 2448 is not smaller than 1991.

**Comment by hschink** [comment]
According to Travis there are still some tests failing.

**Comment by tilt** [comment]
Rebased, passed CI

File: lib/plugins/acts_as_watchable/lib/acts_as_watchable.rb [hunk 2]
```diff
@@ -73,15 +73,26 @@ def self.included(base)
           base.extend ClassMethods
         end
 
-        # Returns an array of users that might be watchers or already are watchers
-        def possible_watcher_users
-          users = User.all
+        def watching_permitted_to_all_users_message
+          "Let all users watch me. If this is unintended, implement :visible? on #{self.class.name}"
```
Comment (hschink) [inline lib/plugins/acts_as_watchable/lib/acts_as_watchable.rb:77]: Shouldn't this go into the localization?

File: spec/models/work_package/work_package_watcher_spec.rb [hunk 3]
```diff
@@ -40,6 +40,10 @@
                                               roles: [role]) }
     let(:work_package) { FactoryGirl.create(:work_package,
                                             project: project) }
+    let!(:user_allowed_to_view_work_packages) do
```
Comment (hschink) [inline spec/models/work_package/work_package_watcher_spec.rb:43]: Why do you add a new user? There is already one in line 33 that can be used by instantiating `project_member`. IMHO we should re-arrange the specs, so we don't need the additional user. What do you think?

File: lib/plugins/acts_as_watchable/lib/acts_as_watchable.rb (hunk 2 above)
Comment (tilt) [inline lib/plugins/acts_as_watchable/lib/acts_as_watchable.rb:77]: Just moved it here, but I assume it should

**Comment by hschink** [comment]
Tested with 48 watchers on a development system: Performance feels reasonable. Thanks! :cherry_blossom:

**Pull request merged**